core/src/main/java/org/apache/carbondata/core/readcommitter/LatestFilesReadCommittedScope.java

```diff
@@ -40,9 +40,26 @@
 public class LatestFilesReadCommittedScope implements ReadCommittedScope {
 
   private String carbonFilePath;
+  private String segmentId;
   private ReadCommittedIndexFileSnapShot readCommittedIndexFileSnapShot;
   private LoadMetadataDetails[] loadMetadataDetails;
 
+  /**
+   * a new constructor of this class, which supports obtain lucene index in search mode
+   *
+   * @param path      carbon file path
+   * @param segmentId segment id
+   */
+  public LatestFilesReadCommittedScope(String path, String segmentId) {
```

Comment:
too much duplicate codes there. It can internally call `this(path)` to reduce the code

Reply:
I don't understand you means.
This method adds segmentId for building indexPath.

Reply:
I mean that these code can be reduced to
```
this(path);
this.segmentId=segmentId;
```
and you should move the next constructor(Line65) above this constructor.

Reply:
ok

Reply:
Can not operate like this, because this(path) has takeCarbonIndexFileSnapShot method, which dependency on segmentId.

Reply:
I change the constructor to   
public LatestFilesReadCommittedScope(String path) {
    this(path, null);
  }